### ISSUE TITLE
canvas fallback text

### ISSUE THREAD

**Issue by jaymollica** [openseadragon/openseadragon#1345]
I'm trying to apply fallback text to the canvas for accessibility purposes:
`<canvas>my fall back text</canvas>`

but it looks like open sea dragon does not close the canvas tag, or have a fallback text option in the documentation?

**Comment by iangilman** [comment]
OSD builds the canvas via JavaScript... is there a way to add the fallback text via JavaScript? Maybe something like `viewer.drawer.canvas.innerHTML = 'my fallback text'`?

Is this for browsers that don't support canvas, or for accessibility for people with screen readers? We already don't use canvas on browsers that don't support it (i.e. IE8, that we no longer officially support that browser).

**Comment by jaymollica** [comment]
It's for people with screen readers. When I do the above, it doesn't look like the fallback text is actually added to the DOM, and the <canvas> element does not have a closing tag in the DOM (see screen shot)
![screen shot 2017-11-20 at 11 50 41 am](https://user-images.githubusercontent.com/4594505/33038369-3f4d3fb6-cde9-11e7-8208-b5fb946a9e91.png)

However, a `console.log(viewer.drawer.canvas.innerHTML)` does print out the fallback text...

**Comment by iangilman** [comment]
Interesting. Does it work with a screen reader when you do that? 

As long as you can add fallback text to a canvas element via JavaScript somehow, you should be able to do it to the OSD canvas.

**Comment by jaymollica** [comment]
It does not work with a screen reader in this case. Is it possible to provide the option for fallback text when you first call Open Sea Dragon?

Canvas is not a great element for accessibility, but it's the curator's preferred method for zoomable images. Getting a fallback for screen readers would be huge for us (and other museums probably). Thanks for your continued attention here!

**Comment by iangilman** [comment]
We can certainly add fallback text, but we'd need to know how to do that. Can you provide an example of a canvas element that's created via JavaScript (not already present in the HTML) that has fallback text that works in a screen reader? 

For reference, we're using `document.createElement('canvas')` to make our canvas. What should we add to that to get the fallback text?

**Comment by jaymollica** [comment]
After a bit of searching around seems like the better method might be to use `myElement.innerHtml("<canvas>my fallback text</canvas>")` but I'm not sure how that would affect the rest of the project.

**Comment by iangilman** [comment]
Fair enough. Well, we can certainly give it a try. Would you be up for writing a patch for it?

**Comment by jaymollica** [comment]
hello, I'll give it a go next month!

**Comment by iangilman** [comment]
Sounds good… Thank you!
